apis/python/src/tiledbsoma/_tdb_handles.py

```diff
@@ -353,14 +352,12 @@ def __repr__(self) -> str:
 def _check_metadata_type(key: str, obj: object) -> None:
     """Pre-checks that a metadata entry can be stored in an array.
 
-    These checks are reproduced from the TileDB Python metadata-setting methods.
-    We have to do this since we don't commit our changes until the very end.
+    These checks are reproduced from the TileDB Python metadata-setting methods,
+    but are slightly more restrictive than what TileDB allows in general.
```

Comment:
```suggestion
    but are slightly more restrictive than what TileDB allows in general. In particular, TileDB disallows arrays
    of string within metadata keys while allowing arrays of int. The SOMA spec, however, disallows arrays
    of any kind. Please also see
    https://github.com/single-cell-data/TileDB-SOMA/blob/main/apis/python/devtools/ingestor#L42-L46
```

Reply:
went with slightly different wording but the same general info